plaso/parsers/powershell_transcript.py

```diff
@@ -0,0 +1,214 @@
+      elif timestamp_pattern.search(transcript[1]):
+        timestamp_string = timestamp_pattern.search(transcript[1])[1]
+        full_headers = False
+      else:
+        parser_mediator.ProduceExtractionWarning('could not find timestamp in '
+          'transcript {0!s} - skipping malformed transcript'.format(transcript))
+        continue
+      # Timestamp format is YYYYMMDDHHmmss
+      time_elements_tuple = int(timestamp_string[:4]),\
+                            int(timestamp_string[4:6]),\
+                            int(timestamp_string[6:8]),\
+                            int(timestamp_string[8:10]),\
+                            int(timestamp_string[10:12]),\
+                            int(timestamp_string[12:])
+      try:
+        start_time = dfdatetime_time_elements.TimeElements(
+          time_elements_tuple=time_elements_tuple)
+      except ValueError:
+        parser_mediator.ProduceExtractionWarning('timestamp \"{0!s}\" seems '
+          'invalid - skipping malformed transcript'.format(time_elements_tuple))
+        continue
+      start_time.is_local_time = True
+      event = time_events.DateTimeValuesEvent(start_time,
+        definitions.TIME_DESCRIPTION_START)
+      # we don't use line breaks here since they aren't displayed nicely in TS
+      if full_headers:
+        command = '; '.join(transcript[18:len(transcript)])
+      else:
+        command = '; '.join(transcript[3:len(transcript)])
+      event_data.command = command
+      # finally create event if values have been set
+      if (event is not None and event_data.command is not None and
+            event_data.command != ''):
+        parser_mediator.ProduceEventWithEventData(event, event_data)
+      else:
+        parser_mediator.ProduceExtractionWarning('skipping transcript {0!s} - '
+          'since relevant event values could not be set'.format(transcript))
+
+
+  def ParseFileObject(self, parser_mediator, file_object):
+    """Parses an PowerShell transcript file-like object.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+      file_object (dfvfs.FileIO): file-like object.
+
+    Raises:
+      WrongParser: when the file cannot be parsed.
+    """
+
+    data = file_object.read(self._HEADER_READ_SIZE)
+    if not data == b'\xef\xbb\xbf**********************':
+      raise errors.WrongParser('Not a PowerShell transcript file')
+
+    # The current offset of the file-like object needs to point at
+    # the start of the file to parse the data correctly.
+    file_object.seek(0, os.SEEK_SET)
+    file_bytes = file_object.read()
+    # If transcript file was opened in editor, line breaks may have been changed
+    if b'\r\n' in file_bytes:
+      line_bytes = file_bytes.split(b'\r\n')
+    else:
+      line_bytes = file_bytes.split(b'\n')
+    lines = [line.decode('utf-8') for line in line_bytes]
+    sep_indices = []
```

Comment:
```sep_indices``` could we choose a more descriptive name for this variable?  Looks like maybe ```newline_indices``` might be a better name for this?

Reply:
Fixed in [1c5ce11](https://github.com/log2timeline/plaso/pull/4168/commits/1c5ce112ec91b6392e2ed80257f6c31c8bd36ca3)